src/dynsight/_internal/analysis/rdf.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from ovito.io import import_file
+from ovito.modifiers import (
+    CoordinationAnalysisModifier,
+    DeleteSelectedModifier,
+    ExpressionSelectionModifier,
+    LoadTrajectoryModifier,
+    TimeAveragingModifier,
+)
+from scipy.signal import find_peaks
+
+
+class RadialDistributionFunction:
+    """Object for computing and analyzing the Radial Distribution Function.
+
+    The Radial Distribution Function (RDF) describes how particle density
+    varies as a function of the distance.
+
+    Parameters:
+        trajectory:
+            List of file paths including trajectory and optional topology file.
+
+        pipeline:
+            OVITO pipeline used for importing and modifying the
+            trajectory data.
+
+        pair_distances:
+            Array of distances between particle pairs.
+
+        rdf:
+            Array of RDF values corresponding to the pair distances.
+
+        minima_points:
+            Tuple containing arrays of minima point distances and RDF values.
+    """
+
+    def __init__(
+        self,
+        trajectory_file: str,
+        topology_file: str | None = None,
+        xyz_cols: list[str] | None = None,
+    ) -> None:
+        """Initialize the RDF object and import the trajectory data.
+
+        Parameters:
+            trajectory_file:
+                Path to the trajectory file (e.g. `.xyz` or `.xtc`).
+
+            topology_file:
+                Path to the topology file if required (e.g. `.gro`).
+
+            xyz_cols:
+                List of column names for XYZ format
+                files (e.g., `["Particle Type", "X", "Y", "Z"]`).
+        """
+        # Check number of args and setup the trajectory files
+        if topology_file is None:
+            self.trajectory = [Path(trajectory_file)]
```

Comment:
Why does this need to be an attribute?

Move to compute, remove self.

I feel like I commented these on a previous PR.